core/src/main/java/org/elasticsearch/search/suggest/term/TermSuggestion.java

```diff
@@ -194,9 +199,27 @@ public void writeTo(StreamOutput out) throws IOException {
             @Override
             protected XContentBuilder innerToXContent(XContentBuilder builder, Params params) throws IOException {
                 builder = super.innerToXContent(builder, params);
-                builder.field(Fields.FREQ, freq);
+                builder.field(Fields.FREQ.getPreferredName(), freq);
                 return builder;
             }
+
+            private static final ConstructingObjectParser<Option, Void> PARSER = new ConstructingObjectParser<>("TermSuggestOptionParser",
```

Comment:
Nit: can we change the name to TermSuggestionOptionParser to reflect the class name?